### PR TITLE
Migrate to Unified Plan/WebRTC 1.0 API

### PR DESCRIPTION
**Pull request by SCG82**

### CODE REVIEW

File: plugins/obs-outputs/WebRTCStream.cpp [hunk 1]
```diff
@@ -273,17 +274,8 @@ bool WebRTCStream::start(Type type)
...
-    //Add the stream to the peer connection
-    if (!pc->AddStream(stream))
-    {
-        //Log
-        error("Adding stream to PeerConnection failed");
-        //Error
-        return false;
-    }
+    //Add video track to peer connection
+    pc->AddTrack(video_track, {"obs"});
```
Comment (agouaillard-cosmo) [inline plugins/obs-outputs/WebRTCStream.cpp]: It s a pity that we lost the capacity to capture an error. Isn't there an equivalent error message with pc->addTrack ?

**Comment by murillo128** [comment]
Why do we need to move to unified plan? we are sending a single stream, imho would be safer to stay in planB

**Comment by agouaillard-cosmo** [comment]
Do you feel strongly against this patch, or is it just a preference?
At one point we will need to move to unified plan anyway, if this patch
does not break anything, I suggest we merge it in.

**Comment by murillo128** [comment]
If it is not broken, don't change it. Unified plan has no benefit over planB for obs and although it should work we should test it and review that there are no regressions.

**Comment by agouaillard-cosmo** [comment]
"if it s no broken don't fix it."
Ok, this is correct, but you know I like "compliance".
I will wait untill I set up automated test and confirm it does not break,
and if does not, I will merge.

File: plugins/obs-outputs/WebRTCStream.cpp (hunk 1 above)
Comment (SCG82) [reply plugins/obs-outputs/WebRTCStream.cpp]: pc->AddTrack has a return type of: RTCErrorOr<rtc::scoped_refptr<RtpSenderInterface>>

**Pull request closed without merging**